From 6692118d6e483ab6b67142e236eda9c4662cba34 Mon Sep 17 00:00:00 2001
From: Tim Harvey <tharvey@gateworks.com>
Date: Wed, 11 Jun 2025 12:00:18 -0700
Subject: [PATCH 2/2] wifi: mac80211: call rate_control_rate_update() for link
 STA

make compatible with upstream commit 88b67e91e292 "(wifi: mac80211: call
rate_control_rate_update() for link STA)"

Signed-off-by: Tim Harvey <tharvey@gateworks.com>
---
 mac.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/mac.c b/mac.c
index 9934778..ae312f7 100644
--- a/mac.c
+++ b/mac.c
@@ -3961,8 +3961,9 @@ void morse_mac_update_ibss_node_capabilities(struct ieee80211_hw *hw,
 /* API to process the bandwidth change notification from mac80211 */
 static void morse_mac_ops_sta_rc_update(struct ieee80211_hw *hw,
 					struct ieee80211_vif *vif,
-					struct ieee80211_sta *sta, u32 changed)
+					struct ieee80211_link_sta *link_sta, u32 changed)
 {
+	struct ieee80211_sta *sta = link_sta->sta;
 	struct morse *mors;
 	struct morse_sta *mors_sta;
 #ifdef CONFIG_MORSE_RC
@@ -4737,7 +4738,7 @@ static struct ieee80211_ops mors_ops = {
 	.tx_last_beacon = morse_mac_ops_tx_last_beacon,
 	.join_ibss = morse_mac_join_ibss,
 	.leave_ibss = morse_mac_leave_ibss,
-	.sta_rc_update = morse_mac_ops_sta_rc_update,
+	.link_sta_rc_update = morse_mac_ops_sta_rc_update,
 	.set_frag_threshold = morse_mac_set_frag_threshold,
 	.set_rts_threshold = morse_mac_set_rts_threshold,
 	.reconfig_complete = morse_mac_reconfig_complete,
-- 
2.34.1

